python/pyarrow/_dataset.pyx

```diff
@@ -2019,9 +2023,12 @@ cdef class Scanner(_Weakrefable):
     dataset : Dataset
         Dataset to scan.
     columns : list of str or dict, default None
-        The columns to project. This can be a list of column names to include
-        (order and duplicates will be preserved), or a dictionary with
-        {new_column_name: expression} values for more advanced projections.
+        The columns to project. This can be a list of column names to
+        include (order and duplicates will be preserved) which may contain the
+        augmented fields such as `batch_index`, `fragment_index`, 
+        `last_in_fragment` and `filename`, or a dictionary
+        with {new_column_name: expression} values for more advanced
+        projections.
```

Comment:
I think elsewhere in this file we have "common" template docstrings to avoid duplication between various wrappers, it may be worth considering for this docstring

Reply:
I am not very sure about the common template doctoring in `_dataset.pyx`.
Do you mean something like this? https://github.com/apache/arrow/blob/4f2e5cc6669aa68dcdf9c876115cf7d1146c8cd5/python/pyarrow/_dataset.pyx#L352

Reply:
Sorry for the late update. I tried following this approach to avoid the duplication. But while setting the docstring using the `__doc__` property, the following error was thrown.
`AttributeError: attribute '__doc__' of 'method_descriptor' objects is not writable`

Reply:
Interesting, it appears this only works for Python classes and not Cython classes. Thanks for checking.